### PR TITLE
validate-modules: Fix some files modules

### PR DESCRIPTION
**Pull request by dagwieers**
##### SUMMARY
This PR includes:
- fixes to validate-modules issues

All modules already include parameter types.

The remaining files-modules have action plugins, so comparing to the arg_spec only is incorrect.

##### ISSUE TYPE
- Docs Pull Request

##### COMPONENT NAME
lineinfile
synchronize

### CODE REVIEW

**Comment by gundalow** [comment]
```
lib/ansible/modules/cloud/misc/ovirt.py:0:0: E309 version_added for new option (instance_gateway) should be '2.8'. Currently StrictVersion ('2.1')
lib/ansible/modules/cloud/misc/proxmox_kvm.py:0:0: E309 version_added for new option (numa_enabled) should be '2.8'. Currently StrictVersion ('0.0')
lib/ansible/modules/cloud/misc/rhevm.py:0:0: E309 version_added for new option (password) should be '2.8'. Currently StrictVersion ('0.0')
test/sanity/validate-modules/ignore.txt:172:1: A102 Remove since "lib/ansible/modules/cloud/misc/ovirt.py" passes "E322" test
test/sanity/validate-modules/ignore.txt:173:1: A102 Remove since "lib/ansible/modules/cloud/misc/ovirt.py" passes "E326" test
test/sanity/validate-modules/ignore.txt:174:1: A102 Remove since "lib/ansible/modules/cloud/misc/proxmox_kvm.py" passes "E322" test
test/sanity/validate-modules/ignore.txt:175:1: A102 Remove since "lib/ansible/modules/cloud/misc/proxmox_kvm.py" passes "E324" test
test/sanity/validate-modules/ignore.txt:176:1: A102 Remove since "lib/ansible/modules/cloud/misc/proxmox_template.py" passes "E323" test
test/sanity/validate-modules/ignore.txt:177:1: A102 Remove since "lib/ansible/modules/cloud/misc/rhevm.py" passes "E322" test
test/sanity/validate-modules/ignore.txt:178:1: A102 Remove since "lib/ansible/modules/cloud/misc/rhevm.py" passes "E324" test
test/sanity/validate-modules/ignore.txt:179:1: A102 Remove since "lib/ansible/modules/cloud/misc/rhevm.py" passes "E335" test
test/sanity/validate-modules/ignore.txt:180:1: A102 Remove since "lib/ansible/modules/cloud/misc/serverless.py" passes "E324" test
test/sanity/validate-modules/ignore.txt:181:1: A102 Remove since "lib/ansible/modules/cloud/misc/terraform.py" passes "E324" test
test/sanity/validate-modules/ignore.txt:182:1: A102 Remove since "lib/ansible/modules/cloud/misc/virt.py" passes "E322" test
test/sanity/validate-modules/ignore.txt:183:1: A102 Remove since "lib/ansible/modules/cloud/misc/virt.py" passes "E326" test
test/sanity/validate-modules/ignore.txt:184:1: A102 Remove since "lib/ansible/modules/cloud/misc/virt_pool.py" passes "E326" test
test/sanity/validate-modules/ignore.txt:400:1: A102 Remove since "lib/ansible/modules/identity/cyberark/cyberark_user.py" passes "E324" test
test/sanity/validate-modules/ignore.txt:401:1: A102 Remove since "lib/ansible/modules/identity/keycloak/keycloak_client.py" passes "E324" test
test/sanity/validate-modules/ignore.txt:402:1: A102 Remove since "lib/ansible/modules/identity/keycloak/keycloak_clienttemplate.py" passes "E324" test
test/sanity/validate-modules/ignore.txt:796:1: A102 Remove since "lib/ansible/modules/source_control/github_issue.py" passes "E324" test
test/sanity/validate-modules/ignore.txt:797:1: A102 Remove since "lib/ansible/modules/source_control/github_issue.py" passes "E326" test
test/sanity/validate-modules/ignore.txt:798:1: A102 Remove since "lib/ansible/modules/source_control/subversion.py" passes "E322" test
```

```
lib/ansible/modules/files/synchronize.py:103:5: key-duplicates DOCUMENTATION: duplication of key "type" in mapping
```

**Comment by dagwieers** [comment]
The CI failure is related to an undocumented parameter that I added.

**Comment by dagwieers** [comment]
We have a fix for the sanity tests for this issue at: #52497

File: lib/ansible/modules/cloud/misc/virt_pool.py [hunk 1]
```diff
@@ -1,100 +1,95 @@
...
-              If 'deleted', pool contents will be deleted and then pool undefined.
+            - Specify which state you want a storage pool to be in.
+            - If C(active), pool will be started.
+            - If C(present), ensure that pool is present but do not change its
+              state; if it is missing, you need to specify xml argument.
+            - If C(inactive), pool will be stopped.
+            - If C(undefined) or C(absent), pool will be removed from I(libvirt) configuration.
+            - If C(deleted), pool contents will be deleted and then pool undefined.
+        type: str
+        choices: [ absent, active,deleted, inactive, present, undefined ]
```
Comment (samccann) [inline lib/ansible/modules/cloud/misc/virt_pool.py]:
```suggestion
        choices: [ absent, active, deleted, inactive, present, undefined ]
```

**Review by samccann — COMMENTED** [review]
one minor nit on the docs portion.

File: lib/ansible/modules/cloud/misc/virt_pool.py (hunk 1 above)
Comment (dagwieers) [reply lib/ansible/modules/cloud/misc/virt_pool.py]: Hmm, I fixed this yesterday?

Comment (dagwieers) [reply lib/ansible/modules/cloud/misc/virt_pool.py]: Maybe I did not push ?

Comment (dagwieers) [reply lib/ansible/modules/cloud/misc/virt_pool.py]: Not sure what happened, but it is fixed now. Thanks ! :-)

**Comment by dagwieers** [comment]
bot_skip

**Comment by dagwieers** [comment]
Because of a mixup between branches, this one included more than just the files-section of changes.
This has now been corrected.

**Review by gundalow — APPROVED** [review]

**Pull request merged**